dwave/optimization/include/dwave-optimization/nodes/mathematical.hpp

```diff
@@ -34,6 +34,11 @@ struct abs {
     constexpr T operator()(const T& x) const { return std::abs(x); }
 };
 
+template <class T>
+struct cos {
```

Comment:
We need these so that they accept any type right?

Reply:
It's because we need `BinaryOp` to be a class, and `std::cos` and `std::sin` are functions.